README.md

```diff
@@ -1,2 +1,164 @@
+
+Example of authoring tools (or "generators") that try to conform to this specification:
+
+- [Bikeshed](http://github.com/tabatkins/bikeshed)
+- [ReSpec](http://github.com/w3c/respec)
+- [Wattsi](https://github.com/whatwg/wattsi)
+
+Note: An accompanying test suite helps assure authoring tools conform to the contract.
+
+## Who consumes these definitions
+
+Aside from specifications internally cross-referencing their own terms and terms in other specifications, "crawlers" extract terms defined in specifications into structured data.
+
+Examples of definition crawlers include:
+
+- [Shepherd](https://dev.csswg.org/projects/shepherd)
+- [Reffy](https://github.com/w3c/reffy)
+
+Crawlers structure and categorized extracted terms into files or databases, allowing for the creation of search tools such as:
+
+- <https://respec.org/xref>
+- Bikeshed's cross-reference search.
+
+## How to get your spec indexed
+
+To have your specification indexed by a crawler, it MUST be registered with:
+
+- [browser specs](https://github.com/w3c/browser-specs/) - See [criteria for inclusion](https://github.com/w3c/browser-specs/blob/master/README.md#spec-selection-criteria).
+- [Shepherd](https://dev.csswg.org/projects/shepherd) - [Contact maintainer(s) directly](https://dev.csswg.org/users/3).
+
+## The Contract
+
+The following sections define the elements, attributes, and attribute values (along with any specific micro-syntax) that constitute the contract.
+
+### Definitions
+
+A <dfn>definition</dfn> is:
+
+- a `<dfn>` element.
+- a `h1`-`h6` element.
+
+No other HTML elements are recognized as defining a term.
+
+#### dfn
+A a dfn-definition MUST have the following attributes:
+
+- `id` attribute, unique to the document.
+- `data-dfn-type`, with one of type values.
+
+The following attributes are OPTIONAL:
+
+- `data-export` or `data-noexport` (they are exclusive).
+- `data-dfn-for`
+
+### Exporting definitions (`data-export`)
+
+Exported definitions MUST include a `data-export` attribute. The attribute's value can be missing or the empty string.
+
+When the `data-export` attribute is present, it means that the definition can be publicly referenced by other documents.
+
+Authors SHOULD only export definitions they intend others to use.
+
+## Explicitly private definitions: (`data-noexport`)
+
+A `data-noexport` attribute means a definition is intended for private use by a specification.
+
+Other specifications MUST NOT reference or link to definitions marked as `data-noexport`.
+
+### Namespacing (`data-dfn-for`)
+
+Definitions can be "for" something.
+
+TODO: The "null" namespace ([= / something =]).
+
+### Types of definitions (`data-dfn-type`)
+
+Every exported definition MUST have a `data-dfn-type`.
+
```

Comment:
it would be good to insert a summary table that for each type indiciates:
* whether it's exported by default or not
* whether it's case sensitive or not
* whether ` data-dfn-for` is required for the said type

Reply:
I definitely agree with this. But for each type, there are specific requirements that I need to work out... like I don't know what `data-dfn-for` for "argument" is right now... I guess it would be "interface/method", but need to check.

Reply:
Yes, that's correct. The `for` is just whatever construct is namespacing the value; arguments clearly are unique only up to their method (in fact, to within one overload of a method).